### PR TITLE
Timezone

### PR DESCRIPTION
**Pull request by hobleyd**
Hey,

I noticed that the Clock wasn't picking up timezone changes; so I fixed that; I also updated the Date handling code to the latest suggested option as well as removing the Enum in favour of a HashMap as I felt it made the code easier to understand.

Cheers,
David

### CODE REVIEW

File: app/src/main/java/com/benny/openlauncher/widget/SearchBar.java [hunk 1]
```diff
@@ -102,6 +111,17 @@ private void init() {
...
+        // These have to match the Preferences Array, but without item 0 as that is a custom option which can be changed:
+        //   <item>@string/custom</item>
+        //   <item>February 17\nSaturday, 2018</item>
+        //   <item>February 17\n15:48</item>
+        //   <item>February 17, 2018\n15:48</item>
+        //   <item>15:48\nFebruary 17, 2018</item>
+        _clockModes.put(1, DateTimeFormatter.ofPattern("MMMM dd\nEEEE, yyyy", Locale.getDefault()));
+        _clockModes.put(2, DateTimeFormatter.ofPattern("MMMM dd\nHH:mm", Locale.getDefault()));
+        _clockModes.put(3, DateTimeFormatter.ofPattern("MMMM dd, yyyy\nHH:mm", Locale.getDefault()));
+        _clockModes.put(4, DateTimeFormatter.ofPattern("HH:mm\nMMMM dd, yyyy", Locale.getDefault()));
```
Comment (dkanada) [inline app/src/main/java/com/benny/openlauncher/widget/SearchBar.java]: I prefer the enum usage but I never touch this class anyway so we can move it to a hash map.

File: app/src/main/java/com/benny/openlauncher/activity/HomeActivity.java [hunk 2]
```diff
@@ -108,15 +109,15 @@ public final void setLauncher(@Nullable HomeActivity v) {
...
 
         _appUpdateIntentFilter.addDataScheme("package");
-        _appUpdateIntentFilter.addAction("android.intent.action.PACKAGE_ADDED");
-        _appUpdateIntentFilter.addAction("android.intent.action.PACKAGE_REMOVED");
-        _appUpdateIntentFilter.addAction("android.intent.action.PACKAGE_CHANGED");
-
+        _appUpdateIntentFilter.addAction(Intent.ACTION_PACKAGE_ADDED);
+        _appUpdateIntentFilter.addAction(Intent.ACTION_PACKAGE_REMOVED);
+        _appUpdateIntentFilter.addAction(Intent.ACTION_PACKAGE_CHANGED);
+
```
Comment (dkanada) [inline app/src/main/java/com/benny/openlauncher/activity/HomeActivity.java]:
```suggestion

```

File: app/src/main/java/com/benny/openlauncher/activity/HomeActivity.java [hunk 3]
```diff
@@ -318,7 +321,7 @@ private void registerBroadcastReceiver() {
         _timeChangedReceiver = new BroadcastReceiver() {
             @Override
             public void onReceive(Context context, Intent intent) {
-                if (intent.getAction().equals(Intent.ACTION_TIME_TICK)) {
+                if (intent.getAction().equals(Intent.ACTION_TIME_TICK) || intent.getAction().equals(Intent.ACTION_TIMEZONE_CHANGED) || intent.getAction().equals(Intent.ACTION_TIME_CHANGED)) {
```
Comment (dkanada) [inline app/src/main/java/com/benny/openlauncher/activity/HomeActivity.java]:
```suggestion
                if (intent.getAction().equals(Intent.ACTION_TIME_TICK)
                    || intent.getAction().equals(Intent.ACTION_TIMEZONE_CHANGED)
                    || intent.getAction().equals(Intent.ACTION_TIME_CHANGED)) {
```

**Comment by hobleyd** [comment]
I have made those changes.

**Comment by Stupidman22** [comment]
ddd MMM partern not work bro ...

**Pull request merged**